File: clojure/filter.clj
(defn get-filter [params id]
  (if (and params (contains? params id))
    (get params id)
    nil))

(defn user-filter [params]
  (get-filter params "user"))

(defn platform-filter [params]
  (get-filter params "platform"))

(defn environment-filter [params]
  (get-filter params "environment"))

(defn course-filter [params]
  (get-filter params "course"))

(defn space-filter [params]
  (get-filter params "space"))

(defn subject-filter [params]
  (get-filter params "subject"))

(defn student-filter [params]
  (get-filter params "student"))

(defn lecture-filter [params]
  (get-filter params "lecture"))

(defn data-filter [params]
  (if (and params (contains? params "start") (contains? params "end"))
    {:start_date (get params "start") :end_date (get params "end")}
    nil))

;; FIXME

File: java/Filter.java
import java.util.Map;

public class Filter {
    private Map<String, String> params;

    public Filter(Map<String, String> params) {
        this.params = params;
    }

    public String filter(String id) {
        if (params != null && params.containsKey(id)) {
            return params.get(id);
        } else {
            return null;
        }
    }

    public String userFilter() {
        return filter("user");
    }

    public String platformFilter() {
        return filter("platform");
    }

    public String environmentFilter() {
        return filter("environment");
    }

    public String courseFilter() {
        return filter("course");
    }

    public String spaceFilter() {
        return filter("space");
    }

    public String subjectFilter() {
        return filter("subject");
    }

    public String studentFilter() {
        return filter("student");
    }

    public String lectureFilter() {
        return filter("lecture");
    }

    public Map<String, String> dataFilter() {
        if (params != null && params.containsKey("start") && params.containsKey("end")) {
            String startDate = params.get("start");
            String endDate = params.get("end");
            
            return Map.of("start_date", startDate, "end_date", endDate);
        } else {
            return null;
        }
    }
}